main/Main.hs

```diff
@@ -171,6 +175,9 @@ main = do
         . setServerName (toS $ "postgrest/" <> prettyVersion) $
         defaultSettings
 
+  when (isLeft socketFileMode) $
+    panic $ (fromJust . leftToMaybe) socketFileMode
```

Comment:
You could use [whenLeft](http://hackage.haskell.org/package/either-5.0.1.1/docs/Data-Either-Combinators.html#v:whenLeft) here(`whenLeft socketFileMode panic`). You'll have to add the `either` lib to the executable in the cabal file but that's fine.